WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/UriWrapper.kt

```diff
@@ -0,0 +1,18 @@
+package org.wordpress.android.ui.posts.mediauploadcompletionprocessors
+
+import android.net.Uri
+
+/**
+ * Wrapper for the Uri class.
+ * 
+ * We need to use the Uri.encode function in order to encode a URL for the VideoPress block.
+ * However, As Uri.encode is part an Android class, it cannot run in unit tests by default.
+ * Note, java.net.URLEncoder is not currently a suitable alternative as it require API 33 and above (our minimum is 24 at the time of writing)
```

Comment:
Another alternative I explored to this approach included "forcing" the unit tests to use an API above 33 then conditionally only using `Uri.encode` for older APIs, while using `java.net.URLEncoder` for newer ones. However, I couldn't find a straightforward way to achieve this. I looked into using a framework like [Robolectric](https://robolectric.org/) but found discussions in which we'd previously decided to remove that framework in https://github.com/wordpress-mobile/WordPress-Android/pull/16170.